src/test/tree/StrategyManangerUnit.tree

```diff
@@ -0,0 +1,89 @@
+StrategyManagerUnit.t.sol
+├── initialize
+│   └── when initialize called again
+│       └── it should revert
+├── depositIntoStrategy()
+│   ├── when deposit successfully
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposit successfully twice
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposits paused
+│   │   └── it should revert
+│   ├── when re-entering
+│   │   └── it should revert
+│   ├── when safeTransferFrom() function reverts
+│   │   └── it should revert
+│   ├── when IERC20 token does not exist
+│   │   └── it should revert
+│   ├── when strategy.deposit() function reverts
+│   │   └── it should revert
+│   ├── when strategy does not exist
+│   │   └── it should revert
+│   ├── when strategy is not whitelisted
+│   │   └── it should revert
+│   └── _addShares()
+│       ├── when shares is 0
+│       │   └── it should revert
+│       └── when deposit would exceed max array length
+│           └── it should revert
+├── depositIntoStrategyWithSignature()
+│   ├── when deposit with signature successfully
+│   │   └── staker nonce should increase and shares should increase
+│   ├── when deposit with signature successfully and replay same deposit signature
+│   │   └── it should revert the second time
+│   ├── when deposit with signature with EIP1271 compliant contract signature
+│   │   └── it should deposit successfully with shares and nonce increase
+│   ├── when deposit with signature with EIP1271 compliant contract but with bad signature
+│   │   └── it should revert
+│   ├── when deposit with signature with non-compliant EIP1271 contract
+│   │   └── it should revert
```

Comment:
this feels a little like its describing tests as opposed to conditions?
there should probably be a branching point for 'given the staker is a contract' vs. 'given the staker is an EOA'

Reply:
You're right, I will rewrite the tree file according to all possible code paths instead and make appropriate changes. 
Will also include the more integration-like scenarios such as increaseDelegatedShares() involving the DelegationManager